kernel/rtlil.h

```diff
@@ -554,6 +554,29 @@ namespace RTLIL
 			return *this;
 		}
 
+		inline ObjIterator<T>& operator+=(int amt) {
+			log_assert(list_p != nullptr);
+			it += amt;
+			if (it == list_p->end()) {
+				(*refcount_p)--;
+				list_p = nullptr;
+				refcount_p = nullptr;
+			}
```

Comment:
@whitequark I'm not sure how this is affected by d71a9badda20d48bcd4031f030d2d0901f837261 .  I guess it doesn't really matter as this is bad behavior anyway, but it would be nice to understand all the implications.

Reply:
Seems fine to me.